packages/components/src/popover/Popover.js

```diff
@@ -1,131 +1,73 @@
 import Types from 'prop-types';
-import classNames from 'classnames';
+import classnames from 'classnames';
+
 import { Position } from '../common';
-import { getPlacement, getPositionRelativeToParent } from './positioning';
-import { wrapInDOMElementIfNecessary } from './DOMWrapping';
-import KeyCodes from '../common/keyCodes';
-import {
-  addClickClassToDocumentOnIos,
-  removeClickClassFromDocumentOnIos,
-} from '../common/domHelpers';
+import ResponsivePanel from '../common/responsivePanel';
 
 import './Popover.css';
 
-export default class Popover extends Component {
-  static Placement = Position;
-
-  constructor() {
-    super();
-    this.state = {
-      isOpen: false,
-    };
-  }
-
-  componentWillUnmount() {
-    this.close();
-  }
-
-  closePopoverOnOutsideClick = (event) => {
-    const popoverClicked = this.popoverElement.contains(event.target);
-    const triggerClicked = this.triggerElement.contains(event.target);
-
-    if (!popoverClicked && !triggerClicked) {
-      this.close();
-    }
-  };
-
-  handleKeyUp = (event) => {
-    if (event.target.nodeName !== 'BUTTON' && event.keyCode === KeyCodes.ENTER) {
-      this.toggle();
-    }
-  };
-
-  open = () => {
-    this.setState({ isOpen: true });
-
-    addClickClassToDocumentOnIos();
-    document.addEventListener('click', this.closePopoverOnOutsideClick, true);
-  };
-
-  close = () => {
-    this.setState({ isOpen: false });
-
-    removeClickClassFromDocumentOnIos();
-    document.removeEventListener('click', this.closePopoverOnOutsideClick, true);
-  };
-
-  toggle = () => (this.state.isOpen ? this.close() : this.open());
-
-  createTrigger = () => {
-    const { children: child, containsFocusableElement } = this.props;
-    const { isOpen } = this.state;
-    const wrappedChild = wrapInDOMElementIfNecessary(child);
-
-    const focusableProps = !containsFocusableElement && {
-      tabIndex: 0,
-      role: 'button',
-      'aria-expanded': isOpen,
-    };
-
-    return cloneElement(wrappedChild, {
-      ...focusableProps,
-      'data-toggle': 'popover',
-      onClick: this.toggle,
-      onKeyUp: this.handleKeyUp,
-      ref: (element) => {
-        this.triggerElement = element;
-      },
-    });
-  };
-
-  style = (className) => this.props.classNames[className] || className;
-
-  render() {
-    const { title, content, preferredPlacement } = this.props;
-    const { isOpen } = this.state;
-
-    const trigger = this.createTrigger();
-
-    const placement = getPlacement(this.popoverElement, preferredPlacement);
-    const popoverClassName = classNames(
-      'tw-popover',
-      this.style('popover'),
-      this.style('animate'),
-      this.style('in'),
-      { [this.style('scale-down')]: !isOpen },
-      this.style(placement),
-    );
+const Popover = ({ children, className, content, preferredPlacement, title }) => {
+  const triggerRef = useRef(null);
+  const [open, setOpen] = useState(false);
+
+  const onClose = (val) => val && setOpen(false);
```

Comment:
```suggestion
  const onClose = () => setOpen(false);
```

can it be like this?